geps/gep-2643/index.md

```diff
@@ -0,0 +1,514 @@
+
+### Two listeners of type TLS and intersecting hostnames
+
+In this case, there is no conflict as the most specific hostname MUST be chosen:
+
+```yaml
+spec:
+  listeners:
+  - hostname: app.user1.tld
+    name: listener1
+    port: 443
+    protocol: TLS
+    tls:
+      mode: Passthrough
+  - hostname: "*.user1.tld"
+    name: listener2
+    port: 443
+    protocol: TLS
+    tls:
+      mode: Passthrough
+status:
+ listeners:
+  - name: listener1
+    conditions:
+    - reason: Accepted
+      status: "True"
+      type: Accepted
+    - reason: NoConflicts
+      status: "False"
+      type: Conflicted
+  - name: listener2
+    conditions:
+    - reason: Accepted
+      status: "True"
+      type: Accepted
+    - reason: NoConflicts
+      status: "False"
+      type: Conflicted
+```
+
+### Two listeners of type TLS with different termination modes
+
+As the support for this feature is `Extended`, in case the implementation
+does not support mixed modes it MUST mark the listeners as conflicted with the 
+reason `ProtocolConflict`:
+
+```yaml
+spec:
+  listeners:
+  - hostname: "*.user2.tld"
+    name: listener1
+    port: 443
+    protocol: TLS
+    tls:
+      mode: Passthrough
+  - hostname: "*.user1.tld"
+    name: listener2
+    port: 443
+    protocol: TLS
+    tls:
+      mode: Terminate
+status:
+ listeners:
+  - name: listener1
+    conditions:
+    - reason: Accepted
+      status: "True"
+      type: Accepted
+    - reason: ProtocolConflict
+      status: "True"
+      type: Conflicted
+  - name: listener2
+    conditions:
+    - reason: Accepted
+      status: "True"
+      type: Accepted
+    - reason: ProtocolConflict
+      status: "True"
+      type: Conflicted
+```
+
+## Hostname intersection between Gateway and TLSRoutes
+
+The following are hostname intersection cases between the Gateway and TLSRoutes
+
+* If a hostname is specified by both the `Listener` and `TLSRoute`, there MUST
+be at least one intersecting hostname for the `TLSRoute` to be attached to the
+`Listener`.  
+  * A `Gateway listener` with `test.example.com` as the hostname matches a `TLSRoute` that
+  specifies at least one of `test.example.com` or `*.example.com`  
+  * A `Gateway listener` with `*.example.com` as the hostname matches a `TLSRoute` that
+  specifies at least one hostname that matches the `Gateway listener` hostname.
+  For example, `test.example.com` and `*.example.com` would both match. On the
+  other hand, `example.com` and `test.example.net` would not match.  
+  * If both the `Gateway listener` and `TLSRoute` specify hostnames, any `TLSRoute`
+  hostnames that do not match the `Gateway listener` hostname MUST be ignored
+  for that Listener. For example, if a `Gateway listener` specified `*.example.com`,
+  and the `TLSRoute` specified `test.example.com` and `test.example.net`,
+  the later must not be considered for a match.  
```

Comment:
It seems as if this scenario should also cause an invalid route condition.  How is the user to know they misconfigured it unless the status condition reflects that?

Reply:
today, if there is only one hostname, it will be marked as "NoMatchingListenerHostname"

but if there are multiple hostnames, there is no condition added. This is the same behavior as HTTPRoute